carlsim/test/compartments.cpp

```diff
@@ -174,6 +174,7 @@ TEST(COMPARTMENTS, spikeTimesCPUvsData) {
 *
 * This test makes sure that CPU mode and GPU mode of the compartment model produce the exact same spike times,
 * even with RK4 at 50 time steps.
+* FIX ME: TEST CURRENTLY FAILS AT numIntSteps = 30 (or 40 depending on machine) with small spike time deviation (2ms)
```

Comment:
BTW, this is the comment I found in CARLsim3. I don't believe the computing results of CPU and GPU would be different given no randomness. Either I overlook some randomness in the test case or there are bugs in your implementation since CARLsim3.
/*!
* \brief testing CARLsim compartments via voltage recording
*
* This test makes sure that firing rate of COBA compartmental groups are similar accross CPU and GPU.
* It also makes sure that soma compartments on CPU and GPU both have similar voltage values every timestep.
*/
/*
TEST(COMPARTMENTS, firingRateCPUvsGPU_COBA_Volt_Rec)